src/Microsoft.Graph.Cli.Core/IO/NativeResponseProcessor.cs

```diff
@@ -0,0 +1,42 @@
+using Microsoft.Kiota.Abstractions;
+
+namespace Microsoft.Graph.Cli.Core.IO;
+
+public class NativeResponseProcessor : IResponseProcessor
+{
+    public async Task<string?> ExtractStringResponseAsync(IResponseHandler responseHandler)
+    {
+        var handler = responseHandler as NativeResponseHandler;
```

Comment:
We could probably write a private helper function that returns an instance of `HttpResponseMessage` or `null` to avoid having to repeat some lines of code.